### PR TITLE
CDAP-4842 add a hydrator-test module containing mock plugins

### PR DESCRIPTION
**Pull request by albertshau**
Adding a hydrator-test module that contains mock plugins used by
the apps for their unit tests. This allows re-use of these mocks
across the three apps, and also allows plugin writers to use them
instead of writing their own mocks.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT3844

File: cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/transform/StringValueFilterTransform.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2016 Cask Data, Inc.
+ * Copyright © 2015 Cask Data, Inc.
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/transform/StringValueFilterTransform.java]: going back a year? :p

File: cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/test/HydratorTestBase.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                      DoubleTransform.class, ErrorTransform.class, IdentityTransform.class,
+                      IntValueFilterTransform.class, StringValueFilterTransform.class);
+  }
+
+  protected static void setupBatchArtifacts(Class<?> appClass) throws Exception {
+    if (startCount++ > 0) {
+      return;
+    }
+
+    // add the artifact for etl batch app
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/test/HydratorTestBase.java]: outdated comment?

Comment (anwar6953) [inline cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/test/HydratorTestBase.java]: might be etl data pipeline app

**Comment by anwar6953** [comment]
2 minor comments; remainder LGTM

**Pull request merged**